e2e/ansible/roles/k8s-openebs-cleanup/tasks/main.yml

```diff
@@ -1,32 +1,63 @@
 ---
 - name: Delete the openebs storage classes
-  shell: source ~/.profile; kubectl delete -f {{ openebs_storageclasses_alias }}
-  args: 
+  shell: source ~/.profile; kubectl delete -f {{ ansible_env.HOME }}/{{ openebs_storageclasses_alias }}
+  args:
+    executable: /bin/bash
+
+- name: Get the list of openebs storage classes deployed from other YAMLs
+  shell: source ~/.profile; kubectl get sc | grep openebs.io/provisioner-iscsi | cut -d ' ' -f1
+  args:
     executable: /bin/bash
+  register: list_sc
+
+- name: Delete storage classes deployed from other YAMLs
+  shell: source ~/.profile; kubectl delete sc {{ item }}
+  args:
+    executable: /bin/bash
+  with_items: "{{ list_sc.stdout_lines }}"
 
 - name: Confirm storage classes have been deleted
   shell: source ~/.profile; kubectl get sc
   args:
     executable: /bin/bash
   register: result
-  until: "'openebs.io/provisioner-iscsi' not in result.stdout" 
+  until: "'openebs.io/provisioner-iscsi' not in result.stdout"
   delay: 10
   retries: 6
 
-- name: Delete the openebs operator 
-  shell: source ~/.profile; kubectl delete -f {{ openebs_operator_alias }}
-  args: 
+- name: Delete the openebs operator
+  shell: source ~/.profile; kubectl delete -f {{ ansible_env.HOME }}/{{ openebs_operator_alias }}
+  args:
     executable: /bin/bash
 
 - name: Confirm pod has been deleted
-  shell: source ~/.profile; kubectl get pods 
+  shell: source ~/.profile; kubectl get pods -n openebs
   args:
     executable: /bin/bash
   register: result
-  until: "'maya-apiserver' or 'openebs-provisioner' not in result.stdout"
+  until: "'maya-apiserver' or 'openebs-provisioner or 'openebs-snapshot-operator' not in result.stdout"
```

Comment:
_n, a ' is missing after openebs-provisioner .

Reply:
Thanks, fixed it.